code/ai/aicode.cpp

```diff
@@ -5846,49 +5852,41 @@ void ai_select_secondary_weapon(object *objp, ship_weapon *swp, int priority1 =
 	num_weapon_types = get_available_secondary_weapons(objp, weapon_id_list, weapon_bank_list);
 
 	// Ignore homing weapons if we didn't specify a flag - for priority 1
-	if ((aip->ai_profile_flags & AIPF_SMART_SECONDARY_WEAPON_SELECTION) && (priority1 == 0)) {
-		ignore_mask |= WIF_HOMING;
-		ignore_mask_without_huge |= WIF_HOMING;
+	if ((aip->ai_profile_flags & AIPF_SMART_SECONDARY_WEAPON_SELECTION) && (prio1.none_set())) {
+		ignore_mask = ignore_mask + Weapon::Info_Flags::Homing_aspect + Weapon::Info_Flags::Homing_heat + Weapon::Info_Flags::Homing_javelin;
+		ignore_mask_without_huge  = ignore_mask_without_huge + Weapon::Info_Flags::Homing_aspect + Weapon::Info_Flags::Homing_heat + Weapon::Info_Flags::Homing_javelin;
```

Comment:
I'm not terribly fond of these two lines (isn't this creating and destroying, like, three temporary flagsets each?); shouldn't it be possible to do something like `ignore_mask.set(Weapon::Info_Flags::Homing_aspect).set(Weapon::Info_Flags::Homing_heat).set(Weapon::Info_Flags::Homing_javelin)`?